README.md

```diff
@@ -1,22 +1,164 @@
-# Working with TypeScript
+# Working with TypeScript Code Along
 
 ## Learning Goals
 
-- a
-- b
-- c
+- Successfully set up a local environment to run TypeScript
+- Configure and understand the tsconfig.json file
+- Practice compiling with `tsc`
 
 ## Introduction
 
-## a
+TypeScript is a very useful and powerful tool we can use to optimize our code by
+checking for syntax errors before runtime, giving tips on how to fix bugs before
+they happen and more, all by adding in types to our regular JavaScript code!
+Though we can start using TypeScript right out of the box, it does recognize
+that every application and developer is different, and gives us some room for
+flexibility. We get access to this flexibility/customization through the
+ts.config file that we will keep in the root of our application, where we can
+change default rules, and add some rules of our own!
 
-## b
+## Environment Setup
 
-## c
+- fork and clone this lab, then open in your text editor
+- install ts globally
+
+You should already have Node Installed on your machine, but if you haven't,
+please do so following the [Node.js Installation Docs][]. If you would like a
+more detailed walk through, please check out the following repositories:
+
+- [Installing NVM & Node.js on MacOS][]
+- [Installing NVM & Node.js on WSL2][]
+
+While we can install TypeScript globally, it's generally best practice to do so
+on a project-by-project basis to avoid different versioning across machines.
+Let's install TypeScript to this project by running the following code in your
+terminal _(Remember to cd into the correct project directory!)_:
+
+```zsh
+$ npm install typescript --save-dev
+```
+
+Once TypeScript is done installing into our project, you should see some
+node_modules files appear into our file directory. Our next step is adding in
+our `tsconfig.json` file.
+
+## The tsconfig.json file
+
+So what _is_ the tsconfig.json file? The tsconfig file is used to help us
+configure and compile our code! This file is _always_ kept in the root of our
+project, and is used to identify the compiler rules we want to enforce, as well
+as specify the root files of the project. The rules the compiler enforces are
+completely customizable and up to you to determine based on your project needs.
+
+There's unlimited ways that you can setup your tsconfig.json file, and a lot of
+different properties you can add to it.
+
+Let's create our tsconfig.json file in the root of our project directory by
+running:
+
+```zsh
+$ touch tsconfig.json
+```
+
+Awesome. Now let's fill in our properties. Were going to start with a
+`"compilerOptions"` key with an object as it's value. The `"compilerOptions"`
```

Comment:
should be its rather than it's (also punctuation wonk)

Reply:
```suggestion
`"compilerOptions"` key with an object as its value. The `"compilerOptions"`
```